### PR TITLE
New Submitit SLURM Launcher plugin

### PR DESCRIPTION
**Pull request by jieru-hu**
## Motivation
port submitit plugin over and we get a slurm launcher!

Major change:
1. update setup.py
2. rename to align with the naming convention 
3. remove the chronos options 
4. Fix imports and misc,  so it works on hydra 1.0
5. Add example app & doc
### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes
## Test Plan

Integration tests

## Related Issues and PRs

https://github.com/facebookresearch/hydra/issues/606

### CODE REVIEW

File: plugins/hydra_submitit_launcher/example/config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# @package _group_
```
Comment (omry) [inline plugins/hydra_submitit_launcher/example/config.yaml:1]: primary config files does not need a package statement (they default to `_global_`).

File: plugins/hydra_submitit_launcher/hydra_plugins/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+__path__ = __import__("pkgutil").extend_path(__path__, __name__)
```
Comment (omry) [inline plugins/hydra_submitit_launcher/hydra_plugins/__init__.py:2]: remove this file, it's no longer needed with Hydra 1.0 (and in fact can cause damage because it's incompatible with namespace packages).

File: plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/conf/hydra/launcher/submitit.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# @package _group_
```
Comment (omry) [inline plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/conf/hydra/launcher/submitit.yaml:1]: This is a good opportunity to port it to use Structured Configs in a followup diff.

File: plugins/hydra_submitit_launcher/setup.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from setuptools import setup, find_namespace_packages
+
+with open("README.md", "r") as fh:
+    LONG_DESC = fh.read()
+    setup(
+        name="hydra-submitit-launcher",
+        version="0.1.10",
```
Comment (omry) [inline plugins/hydra_submitit_launcher/setup.py:8]: update version to 1.0.0rc1

File: website/docs/plugins/submitit_launcher.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
+---
+id: submitit_launcher
+title: Submitit Launcher plugin
+sidebar_label: Submitit Launcher plugin
+---
+
+The Submitit Launcher plugin provides a launcher for tasks based on [`Submitit`](https://github.com/facebookincubator/submitit).
```
Comment (omry) [inline website/docs/plugins/submitit_launcher.md:7]:
```suggestion
The Submitit Launcher plugin provides a [SLURM ](https://slurm.schedmd.com/documentation.html)Launcher based on [`Submitit`](https://github.com/facebookincubator/submitit).
```

Comment (omry) [inline website/docs/plugins/submitit_launcher.md:7]: Please align with other plugins readme header.

File: plugins/hydra_submitit_launcher/setup.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        author="Omry Yadan",
+        author_email="omry@fb.com",
+        description="Submitit plugin for Hydra",
```
Comment (omry) [inline plugins/hydra_submitit_launcher/setup.py:11]:
```suggestion
        description="Submitit Hydra Launcher plugin",
```

File: plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/conf/hydra/launcher/submitit.yaml (hunk 3 above)
Comment (jieru-hu) [reply plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/conf/hydra/launcher/submitit.yaml:1]: Create https://github.com/facebookresearch/hydra/issues/623 for this.

File: plugins/hydra_joblib_launcher/setup.py [hunk 7]
```diff
@@ -19,7 +19,6 @@
             "Programming Language :: Python :: 3.7",
             "Programming Language :: Python :: 3.8",
             "Operating System :: MacOS",
-            "Operating System :: Microsoft :: Windows",
```
Comment (omry) [inline plugins/hydra_joblib_launcher/setup.py]: Why did you remove Windows support from the Joblib launcher?

File: plugins/hydra_submitit_launcher/README.md [hunk 8]
```diff
@@ -0,0 +1 @@
+# Hydra Submitit plugin
```
Comment (omry) [inline plugins/hydra_submitit_launcher/README.md:1]: Make a pass comparing this plugin name, package name module names etc to the other plugins to be sure it's consistent.

File: plugins/hydra_submitit_launcher/setup.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from setuptools import find_namespace_packages, setup
+
+with open("README.md", "r") as fh:
+    LONG_DESC = fh.read()
+    setup(
+        name="hydra-submitit-launcher",
+        version="1.0.0rc1",
+        author="Omry Yadan",
+        author_email="omry@fb.com",
```
Comment (omry) [inline plugins/hydra_submitit_launcher/setup.py:10]: Add your name/email to the list!

File: plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+    def checkpoint(self, *args, **kwargs):
+        """Resubmit the current callable at its current state with the same initial arguments."""
+        # lazy import to ensure plugin discovery remains fast
+        import submitit
+
+        return submitit.helpers.DelayedSubmission(self, *args, **kwargs)
+
+    def launch(
+        self, job_overrides: Sequence[Sequence[str]], initial_job_idx: int
```
Comment (omry) [inline plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py:88]: You added initial_job_idx, right?
You also need to use it.

This also suggests that you missed a test. Look at the example launcher and be sure to introduce the tests from there.

**Comment by omry** [comment]
Hurray!
